airflow/models/dag.py

```diff
@@ -1753,6 +1756,44 @@ def get_last_dagrun(self, session=None, include_externally_triggered=False):
     def safe_dag_id(self):
         return self.dag_id.replace('.', '__dot__')
 
+    @property
+    def code(self):
+        if conf.getboolean('core', 'store_code', fallback=False):
+            return self._get_code_from_db(self.fileloc)
+        else:
+            return self._get_code_from_file(self.fileloc)
+
+    def _get_code_from_file(self, fileloc):
+        with self._open_maybe_zipped(fileloc, 'r') as f:
+            code = f.read()
+        return code
+
+    @classmethod
+    def _open_maybe_zipped(cls, f, mode='r'):
+        """
+        Opens the given file. If the path contains a folder with a .zip suffix, then
+        the folder is treated as a zip archive, opening the file inside the archive.
+
+        :return: a file object, as in `open`, or as in `ZipFile.open`.
+        """
+        ZIP_REGEX = re.compile(r'((.*\.zip){})?(.*)'.format(re.escape(os.sep)))
+        _, archive, filename = ZIP_REGEX.search(f).groups()
+        if archive and zipfile.is_zipfile(archive):
+            return zipfile.ZipFile(archive, mode=mode).open(filename)
+        else:
+            return io.open(f, mode=mode)
+
+    @provide_session
+    def _get_code_from_db(self, fileloc, session=None):
```

Comment:
```suggestion
    def _get_code_from_db(self, session=None):
```

Since this is a method on self we shouldn't pass in the property but instead just look at `self.fileloc`